modules/programs/tarsnap.nix

```diff
@@ -0,0 +1,300 @@
+          this directory will not be backed up by tarsnap, so it should not be
+          used for any other purpose. If the directory cache-dir is lost, it can
+          be reconstructed by running <command>tarsnap --fsck</command>.
+        '';
+      };
+
+      checkpointBytes = mkOption {
+        type = types.str;
+        default = "1G";
+        description = ''
+          Create a checkpoint after every bytespercheckpoint bytes of uploaded
+          data. The value bytespercheckpoint must be at least 1000000, and a
+          higher value is recommended since creating a checkpoint in an archive
+          can take a few seconds and several hundred kB of bandwidth.
+        '';
+      };
+
+      diskPause = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Pause for X ms between storing archive entries and after every 64 kB
+          of file data. This will slow down tarsnap and thereby reduce its
+          impact on other applications. For archiving files which are stored on
+          an ATA disk and are not in the operating system disk cache, a value of
+          diskPause = 10 will approximately double the time taken.
+        '';
+      };
+
+      exclude = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Do not process files or directories that match the specified pattern.
+          Note that exclusions take precedence over patterns or filenames
+          specified on the command line.
+        '';
+      };
+
+      forceResources = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Force the decryption of a passphrase-encrypted key file to proceed
+          even if it is anticipated to require an excessive amount of memory or
+          CPU time
+        '';
+      };
+
+      humanizeNumbers = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Use SI prefixes to make numbers printed by --print-stats and SIGINFO
+          more readable.
+        '';
+      };
+
+      include = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Process only files or directories that match the specified pattern.
+          Note that exclusions specified with exclude take precedence over
+          inclusions. If no inclusions are explicitly specified, all entries are
+          processed by default. The --include option is especially useful when
+          filtering archives. For example, the command <command>tarsnap -c -f
+          foo-backup --include='*foo*' @@all-backup</command> creates a new
+          archive foo-backup containing only the entries from all-backup
+          containing the string Sq foo.
+        '';
+      };
+
+      insaneFilesystems = mkOption {
```

Comment:
Would anybody actually use this option? Better remove it, anybody wanting to use it can supply it in `extraConfig`.

Reply:
sure.